load_eessi_extend_module.sh

```diff
@@ -0,0 +1,116 @@
+# Assumptions (if one is not satisfied the script prints a message and exits)
+# - EESSI version is given as first argument
+# - TMPDIR is set
+# - EB is set
+# - EASYBUILD_INSTALLPATH needs to be set
+# - Function check_exit_code is defined;
+#   scripts/utils.sh in EESSI/software-layer repository defines this function, hence
+#   scripts/utils.sh shall be sourced before this script is run
+#
+# This script is part of the EESSI software layer, see
+# https://github.com/EESSI/software-layer.git
+#
+# author: Kenneth Hoste (@boegel, HPC-UGent)
+# author: Alan O'Cais (@ocaisa, CECAM)
+# author: Thomas Roeblitz (@trz42, University of Bergen)
+#
+# license: GPLv2
+#
+#
+set -o pipefail
+
+# this script is *sourced*, not executed, so can't rely on $0 to determine path to self or script name
+# $BASH_SOURCE points to correct path or script name, see also http://mywiki.wooledge.org/BashFAQ/028
+if [ $# -ne 1 ]; then
+    echo "Usage: source ${BASH_SOURCE} <EESSI-extend version>" >&2
+    exit 1
+fi
+
+EESSI_EXTEND_VERSION="${1}-easybuild"
+
+# make sure that environment variables that we expect to be set are indeed set
+if [ -z "${TMPDIR}" ]; then
+    echo "\$TMPDIR is not set; exiting" >&2
+    exit 2
+fi
+
+# ${EB} is used to specify which 'eb' command should be used;
+# can potentially be more than just 'eb', for example when using 'eb --optarch=GENERIC'
+if [ -z "${EB}" ]; then
+    echo "\$EB is not set; exiting" >&2
+    exit 2
+fi
+
+# ${EASYBUILD_INSTALLPATH} points to the installation path and needs to be set
+if [ -z "${EASYBUILD_INSTALLPATH}" ]; then
+    echo "\$EASYBUILD_INSTALLPATH is not set; exiting" >&2
+    exit 2
+fi
+
+# make sure that utility functions are defined (cfr. scripts/utils.sh script in EESSI/software-layer repo)
+type check_exit_code
+if [ $? -ne 0 ]; then
+    echo "check_exit_code function is not defined; exiting" >&2
+    exit 3
+fi
+
+echo ">> Checking for EESSI-extend module..."
+
+ml_av_eessi_extend_out=${TMPDIR}/ml_av_eessi_extend.out
+# need to use --ignore_cache to avoid the case that the module was removed (to be
+# rebuilt) but it is still in the cache
+module --ignore_cache avail 2>&1 | grep -i EESSI-extend/${EESSI_EXTEND_VERSION} &> ${ml_av_eessi_extend_out}
+
+if [[ $? -eq 0 ]]; then
+    echo_green ">> Module for EESSI-extend/${EESSI_EXTEND_VERSION} found!"
+else
+    echo_yellow ">> No module yet for EESSI-extend/${EESSI_EXTEND_VERSION}, installing it..."
+
+    EB_TMPDIR=${TMPDIR}/ebtmp
+    echo ">> Using temporary installation of EasyBuild (in ${EB_TMPDIR})..."
+    pip_install_out=${TMPDIR}/pip_install.out
+    pip3 install --prefix ${EB_TMPDIR} easybuild &> ${pip_install_out}
+
+    # keep track of original $PATH and $PYTHONPATH values, so we can restore them
+    ORIG_PATH=${PATH}
+    ORIG_PYTHONPATH=${PYTHONPATH}
+
+    # source configure_easybuild to use correct eb settings
+    (
+        EESSI_MAIN_DIR=$(dirname $(readlink -f $BASH_SOURCE))
+        source ${EESSI_MAIN_DIR}/configure_easybuild
```

Comment:
Is this really necessary? This is forcing us to maintain both `EESSI-extend` _and_ `configure_easybuild`

Reply:
How about
```bash
export EASYBUILD_READ_ONLY_INSTALLDIR=1
```

Reply:
Plus maybe
```bash
export EASYBUILD_EXPERIMENTAL=1
```
if we were ever going to install it via an easystack file.

I'll add these three, but comment out the last.